test/js/source/vector_tile_source.test.js

```diff
@@ -112,6 +112,34 @@ test('VectorTileSource', function(t) {
     testScheme('xyz', 'http://example.com/10/5/5.png');
     testScheme('tms', 'http://example.com/10/5/1018.png');
 
+    t.test('reloads a loading tile properly', function (t) {
+        var source = createSource({
+            tiles: ["http://example.com/{z}/{x}/{y}.png"]
+        });
+        var events = [];
+        source.dispatcher.send = function(type, params, cb) {
+            events.push(type);
+            setTimeout(cb, 0);
+            return 1;
+        };
+
+        source.on('load', function () {
+            var tile = {
+                coord: new TileCoord(10, 5, 5, 0),
+                state: 'loading',
+                loadVectorData: function () {
+                    this.state = 'loaded';
+                    events.push('tile loaded');
+                }
+            };
+            source.loadTile(tile, function () {});
```

Comment:
Just to be extra explicit, how about asserting `t.same(tile.state, 'loading')` after the first `loadTile` call?  The `setTimeout` in dispatcher.send ensures this, but I think including the assertion would make the test code a better expression of what we're trying to cover here